src/components/SiteFooter/index.jsx

```diff
@@ -0,0 +1,214 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Hyperlink, Icon } from '@edx/paragon';
+
+class SiteFooter extends React.Component {
+  renderSiteLogo() {
+    return (
+      <img src={this.props.siteLogo} alt={`${this.props.siteName} logo`} />
+    );
+  }
+
+  renderMarketingSiteUrl(path) {
+    return `${this.props.marketingSiteBaseUrl}${path}`;
+  }
+
+  renderSocialLinks() {
+    const {
+      siteName,
+      showSocialLinks,
+      facebookUrl,
+      twitterUrl,
+      youTubeUrl,
+      linkedInUrl,
+      googlePlusUrl,
+      redditUrl,
+    } = this.props;
+    let socialLinks = null;
+    if (showSocialLinks) {
+      socialLinks = (
```

Comment:
Had you considered a design pattern where a full list of link names and URLs came in as properties?  Perhaps this implementation is sufficient for edX white-label sites if all white-label sites have these same set of links.  However, eventually, for Open edX, we'll need another implementation that allows the app to pass in its own set of links - as it may have a different set of social integrations.

Reply:
Not sure how that would work. What did you have in mind?

Reply:
You could accept a (props) socialLinks array of objects, something like this?

```
// to be provided via props
const socialLinks = [
  {
    title: "Facebook",
    url: "http://facebook.com",
    iconClassName: 'fa fa-something',
    screenReaderText: 'Like us on Facebook'
  },
  {
    title: "Twitter",
    url: "http://twitter.com",
    iconClassName: 'fa fa-something',
    screenReaderText: 'Follow us on Twitter'
  },

];

return this.props.socialLinks.map(function(socialLink) {
  return (
    <li>
        <a href={socialLink.url} title={socialLink.title} rel="noopener noreferrer" target="_blank">
        <Icon className={socialLink.iconClassName} screenReaderText={socialLink.screenReaderText} />
      </a>
    </li>
  )
})
```